src/main/java/com/fasterxml/jackson/datatype/joda/cfg/JacksonJodaDateFormat.java

```diff
@@ -251,6 +266,11 @@ public DateTimeFormatter createFormatter(SerializerProvider ctxt)
                 formatter = formatter.withZone(DateTimeZone.forTimeZone(tz));
             }
         }
+        if (!ctxt.isEnabled(SerializationFeature.WRITE_DATES_WITH_CONTEXT_TIME_ZONE)) {
+            if (valueTimeZone != null && !valueTimeZone.equals(_jdkTimezone)) {
```

Comment:
I don't think this can ever be true as `valueTimeZone` is Joda's `DateTimeZone` and `_jdkTimeZone` is JDK `java.util.TimeZone` (IDEA flags this as likely flaw).
Should maybe be:

    if (valueTimeZone != null && !valueTimeZone.toTimeZone().equals(_jdkTimezone)) {

?

Oh, also maybe `null` check for `_jdkTimeZone`,

    if (valueTimeZone != null) &&
      ((_jdkTimeZone == null) || !valueTimeZone.toTimeZone().equals(_jdkTimezone))) {

Reply:
Yes, make sense 👍🏼